internal/ingress/controller/parser/parser.go

```diff
@@ -560,9 +565,36 @@ func overrideService(service *Service,
 	}
 }
 
-func overrideRoute(route *Route,
+// overrideServiceByAnnotation sets the Service protocol via annotation
+func overrideServiceByAnnotation(service *Service,
+	anns map[string]string) {
+	protocol := annotations.ExtractProtocolName(anns)
+	if len(protocol) == 0 {
+		return
+	}
+	service.Protocol = kong.String(protocol)
```

Comment:
Sanitize the protocol; if it is not valid, then don't override it.
Same comment for route overriding logic as well.